### DISCUSSION TITLE
How to make the swagger all request to attach Authorization request header without @RolesAllowed?

### DISCUSSION THREAD

**Discussion by seepine** [quarkusio/quarkus#31413] [Q&A]
## Pre work
```kts
implementation("io.quarkus:quarkus-smallrye-openapi")
```

```properties
quarkus.smallrye-openapi.security-scheme=jwt
```
## And then
And open `http://127.0.0.1:8080/q/swagger-ui/`, i got it
![image](https://user-images.githubusercontent.com/93666053/221343107-94f0bcba-a74a-4a71-af1a-164f8c03b7dd.png)

But i request api , not find Authorization request header, and swagger-ui not `lock` on the right
![image](https://user-images.githubusercontent.com/93666053/221343286-8ebfecfe-22c2-40a7-ac76-d838c13e03e1.png)

Unless add `@RolesAllowed` it can be work, and have lock on the right in swagger-ui
![image](https://user-images.githubusercontent.com/93666053/221343254-19db4e5e-677b-4ce8-b666-b984bc3e377f.png)

## Question
How to make the swagger all request to attach Authorization request header, without `@RolesAllowed`? Because I use other authentication frameworks, not jwt

[a comment by an automated account is not shown]

**Reply by phillip-kruger** [reply]
`@RolesAllowed` is not part of JWT. It just define what roles are allowed.

**Reply by seepine** [reply]
> `@RolesAllowed` is not part of JWT. It just define what roles are allowed.

thx, back to the question, how to make the swagger all request to attach `Authorization` request header, without `@RolesAllowed`?

**Reply by phillip-kruger** [reply]
You can use the OpenAPI Annotations. So SecurityScheme and SecurityRequirement.  So something like `@SecurityRequirement(name = "Authorization")` on the methods and `@SecurityScheme(securitySchemeName = "Authorization",
        ...)` on the ApplicationConfig or the class
